ocs_ci/ocs/ui/base_ui.py

```diff
@@ -0,0 +1,176 @@
+        )
+        os.mkdir(self.screenshots_folder)
+        logger.info(f"screenshots pictures:{self.screenshots_folder}")
+        self.cnt_screenshot = 0
+
+    def do_click(self, by_locator, type=By.XPATH, timeout=30):
+        """
+        Click on Button/link on OpenShift Console
+
+        by_locator (str): GUI element needs to operate on.
+        type (By): Set of supported locator strategies.
+        timeout (int): Looks for a web element repeatedly until timeout (sec) happens.
+
+        """
+        screenshot = ocsci_config.UI_SELENIUM.get("screenshot")
+        if screenshot:
+            self.take_screenshot()
+        wait = WebDriverWait(self.driver, timeout)
+        element = wait.until(ec.element_to_be_clickable((type, by_locator)))
+        element.click()
+
+    def do_send_keys(self, by_locator, text, type=By.XPATH, timeout=30):
+        """
+        Send text to element on OpenShift Console
+
+        by_locator (str): GUI element needs to operate on.
+        text (str): Send text to element
+        type (By): Set of supported locator strategies
+        timeout (int): Looks for a web element repeatedly until timeout (sec) happens.
+
+        """
+        wait = WebDriverWait(self.driver, timeout)
+        element = wait.until(ec.element_to_be_clickable((type, by_locator)))
+        element.send_keys(text)
+        screenshot = ocsci_config.UI_SELENIUM.get("screenshot")
+        if screenshot:
+            self.take_screenshot()
+
+    def is_expanded(self, by_locator, type=By.XPATH, timeout=30):
+        """
+        Check whether an element is in an expanded or collapsed state
+
+        Args:
+            by_locator (str): GUI element needs to operate on.
+            type (By): Set of supported locator strategies
+            timeout (int): Looks for a web element repeatedly until timeout (sec) happens.
+
+        return:
+            bool: True if element expended, False otherwise
+
+        """
+        wait = WebDriverWait(self.driver, timeout)
+        element = wait.until(ec.element_to_be_clickable((type, by_locator)))
+        return True if element.get_attribute("aria-expanded") == "true" else False
+
+    def choose_expanded_mode(self, mode, by_locator, type):
+        """
+        Select the element mode (expanded or collapsed)
+
+        mode (bool): True if element expended, False otherwise
+        by_locator (str): GUI element needs to operate on.
+        type (By): Set of supported locator strategies
+
+        """
+        current_mode = self.is_expanded(by_locator=by_locator, type=type)
+        if mode != current_mode:
+            self.do_click(by_locator=by_locator, type=type)
+
+    def take_screenshot(self):
+        """
+        Take screenshot using python code
+
+        """
+        time.sleep(1)
+        self.cnt_screenshot += 1
+        filename = os.path.join(
+            self.screenshots_folder, f"{self.cnt_screenshot:03}.png"
```

Comment:
Now when I'm looking at the last validation [test execution](https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/16512/console) log and the list of [created screenshots](http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/jnk-pr3661-b3779/jnk-pr3661-b3779_20210119T224154/logs/screenshots_ui_1611100406/), I'll suggest to construct the screenshot name not from incremental number, but also include date - so it will be easier to distinguish which screenshot belongs to which step in the log.
Also it might be worth to print some log message for each screenshot (maybe in debug level) - something like: `Creating snapshot: <snapshot/path>`

Reply:
@dahorak I changed the file name of screenshot and added a log (per sceenshot).
Is Ok?

Reply:
Yep